server/src/models/Vote.js

```diff
@@ -0,0 +1,43 @@
+const Model = require("./Model.js")
+
+class Vote extends Model {
+  static get tableName() {
+    return "votes"
+  }
+  static get jsonSchema() {
+    return {
+      type: "object",
+      required: ["userId","characterId","voteValue"],
```

Comment:
Careful! We actually don't want to validate these foreign keys as required because we would not be able to use `$relatedQuery` to insert new records. When using relatedQuery, we call on an instance of an object, and when used with `insert` it will create the new object, and use the original instance as a reference association

```js
const character = await Character.query().findById(1)

const newReview = character.$relatedQuery("reviews").insertAndFetch({ ... })
```

With the above syntax, we don't need to supply the `characterId` within the `insertAndFetch` because it's able to use the `character` instance object to build the association.

The current jsonSchema here would not allow us to run this query, because the model is expecting both a `characterId` and `userId` to be in the `insert` method

Reply:
In the case of the Vote model and table, we are not returning anything from the table other than the sum of a character's votes, and our vote router is inserting/updating/deleting votes using only the Vote model.

Reply:
In the now-existing charactersVotesRouter, we are grabbing the character ID and user ID from params and req.user, respectively.  Those are being used in conjunction with the voteValue, from the front end, to create the insert query.